# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/ViewportUtils.h
# Commit: be8f5eb58460
# Full Hash: be8f5eb58460dc98673c9890caaf2cf00a3271ad
# Author: Botond Ballo <botond@mozilla.com>
# Date: 2020-04-28 16:30:11
# Regressor Bug: 1556556
# File Overlap Count: 1
# Description:
#   Bug 1556556 - Add some VisualToLayout() and LayoutToVisual() wrappers to ViewportUtils. r=kats
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68724
# ==============================================================================

diff -r 629c58a9166b -r be8f5eb58460 layout/base/ViewportUtils.h
--- a/layout/base/ViewportUtils.h	Tue Apr 28 01:35:18 2020 +0000
+++ b/layout/base/ViewportUtils.h	Tue Apr 28 01:36:22 2020 +0000
@@ -8,8 +8,12 @@
 #include "Units.h"
 #include "mozilla/layers/ScrollableLayerGuid.h"
 
+class nsIFrame;
+
 namespace mozilla {
 
+class PresShell;
+
 class ViewportUtils {
  public:
   /* Return a transform to be applied to the coordinates of input events
@@ -32,6 +36,19 @@
        documentation for additional details. */
   static CSSToCSSMatrix4x4 GetVisualToLayoutTransform(
       layers::ScrollableLayerGuid::ViewID aScrollId);
+
+  /* The functions below apply GetVisualToLayoutTransform() or its inverse
+   * to various quantities.
+   *
+   * To determine the appropriate scroll id to pass into
+   * GetVisualToLayoutTransform(), these functions prefer to use the one
+   * from InputAPZContext::GetTargetLayerGuid() if one is available.
+   * If one is not available, they use the scroll id of the root scroll
+   * frame of the pres shell passed in as |aContext| as a fallback.
+   */
+  static nsPoint VisualToLayout(const nsPoint& aPt, PresShell* aContext);
+  static nsRect VisualToLayout(const nsRect& aRect, PresShell* aContext);
+  static nsPoint LayoutToVisual(const nsPoint& aPt, PresShell* aContext);
 };
 
 }  // namespace mozilla
